config/confighttp/compression.go

```diff
@@ -146,6 +147,7 @@ func (d *decompressor) wrap(h http.Handler) http.Handler {
 	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
 		newBody, err := newBodyReader(r)
 		if err != nil {
+			r.Body.Close()
```

Comment:
This is a separate issue, would be nice if we fix this in a separate PR.

Reply:
as @atoulme  said, this pr would also test the compression reader close the real http.body.
so i add a closeOnceReader in TestHTTPContentDecompressionHandler.
and i need line 150 to pass the test.
![截屏2023-05-15 上午10 10 48](https://github.com/open-telemetry/opentelemetry-collector/assets/52686065/46694a50-d51a-4043-812c-5d8da8be796f)

Reply:
But I don't see any check that it is really closed (in the tests). I do see something that you return error on multiple calls (but the return error is ignored on line 150 and 155 so that check is not useful.

Reply:
> But I don't see any check that it is really closed (in the tests). I do see something that you return error on multiple calls (but the return error is ignored on line 150 and 155 so that check is not useful.

in compression_test, i implement a closeOnceReader, which would return net.ErrClosed when closed multiple time.
and in line 269, i check the real http.body is already closed whenever the compression reader create successfully or not.